### PR TITLE
feat: solidity version upgrade

### PR DESCRIPTION
**Pull request by Jemiiah**
resolve issue : #11

### CODE REVIEW

File: index.ts
Comment (kfastov) [inline index.ts:1]: Why do we need this?

File: package.json [hunk 1]
```diff
@@ -45,5 +46,10 @@
...
     "test:coverage": "forge coverage",
     "test:coverage:report": "forge coverage --report lcov && genhtml lcov.info --branch-coverage --output-dir coverage"
+  },
+  "module": "index.ts",
+  "type": "module",
+  "peerDependencies": {
+    "typescript": "^5.0.0"
   }
-}
+}
```
Comment (kfastov) [inline package.json:55]: Again, do we really need this here? Did you have any difficulties with running Bun?

File: src/LineaStateBridge.sol [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.15;
+pragma solidity ^0.8.19;
```
Comment (kfastov) [inline src/LineaStateBridge.sol:2]: If we specify the concrete single Solidity version in foundry config, we may use "=" instead of "^", if it wouldn't cause any problems. What do you think?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "strict": true,
+    "skipLibCheck": true,
+    "noFallthroughCasesInSwitch": true,
+
+    // Some stricter flags (disabled by default)
+    "noUnusedLocals": false,
+    "noUnusedParameters": false,
+    "noPropertyAccessFromIndexSignature": false
+  }
+}
```
Comment (kfastov) [inline tsconfig.json:27]: It's fine if you leave this, just tell in the comments why did you add this config and picked these concrete values 😊

File: package.json (hunk 1 above)
Comment (Jemiiah) [reply package.json:55]: Yes I had difficulties running bun

File: index.ts
Comment (Jemiiah) [reply index.ts:1]: I had difficulties running bun and I had to install different dependencies for it run

File: src/LineaStateBridge.sol (hunk 2 above)
Comment (Jemiiah) [reply src/LineaStateBridge.sol:2]: we could use 0.8.19 like that it wouldn't give any issue

**Comment by kfastov** [comment]
@Jemiiah Please remove the unrelated changes (package.json, index.ts, tsconfig.json. If you know some best practices that can be utilized in this project, you can open a separate PR.
